fuzz/C++/fuzz_XMLProfiles/fuzz_XMLProfiles.cxx

```diff
@@ -23,18 +22,7 @@ extern "C" int LLVMFuzzerTestOneInput(
         return EXIT_FAILURE;
     }
 
-    const char* filename = buf_to_file(data, size);
-
-    if (filename == NULL)
-    {
-        return EXIT_FAILURE;
-    }
-
-    // TODO change this to a func. taking buf + len (or C string)
-    // to avoid using `buf_to_file`
-    xmlparser::XMLProfileManager::loadXMLFile(filename);
-
-    if (delete_file(filename) != 0)
+    if (ReturnCode_t::RETCODE_OK != fastdds::dds::DomainParticipantFactory::get_instance()->load_XML_profiles_string(reinterpret_cast<const char*>(data), size))
     {
         return EXIT_FAILURE;
     }
```

Comment:
I don't think we need to return with failure if the parser correctly identified an error
```suggestion
    fastdds::dds::DomainParticipantFactory::get_instance()->load_XML_profiles_string(reinterpret_cast<const char*>(data), size)
```

Reply:
The logic is the same as before. I think it makes sense that `LLVMFuzzerTestOneInput()` return an error if it is not able to load or parse the XML. Do you think would it be better if we return ERROR only if `==RETCODE_ERROR`? In all the other cases we would still fail in parsing the XML, but it wouldn't return as error.

Reply:
The behavior has changed. Previously, the return of `xmlparser::XMLProfileManager::loadXMLFile()` was not being checked. The purpose of fuzzers is fail if and only if an uncontrolled situation is reached (i.e. unhandled exception, access violations, etc.). They will be constantly trying with wrong inputs to check if the library crashes. If `load_XML_profiles_string` returns then the fuzzer succeeded, no matter which code it is returning

Reply:
It actually returned error if there was an error in creating the file or in deleting it. Ok, if you think it's outside the scope of the method informing about parsing failure, I will always return true